extensions/file/src/main/java/org/mule/extension/file/internal/DirectoryListener.java

```diff
@@ -220,12 +225,12 @@ public void onStart(SourceCallback<InputStream, ListenerFileAttributes> sourceCa
     createWatcherService();
 
     matcher = predicateBuilder != null ? predicateBuilder.build() : new NullFilePayloadPredicate();
-    executorService =
-        newSingleThreadExecutor(r -> new Thread(r,
-                                                format("%s%s.file.listener", getPrefix(muleContext), flowConstruct.getName())));
+    // TODO MULE-11018 format("%s%s.file.listener", getPrefix(muleContext), flowConstruct.getName())
+    scheduler = schedulerService.ioScheduler();
+
     started = true;
     stopRequested.set(false);
-    executorService.execute(() -> listen(sourceCallback));
+    scheduledListen = scheduler.scheduleWithFixedDelay(() -> listen(sourceCallback), 0, 1, MILLISECONDS);
```

Comment:
this cannot be a recurring task. This HAS to be a single threaded loop as it was before